gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -482,51 +472,50 @@ impl GatewayLightning for ClnRpcService {
         Ok(tonic::Response::new(outcome))
     }
 
-    type RouteHtlcsStream = ReceiverStream<Result<RouteHtlcResponse, Status>>;
+    type RouteHtlcsStream = ReceiverStream<Result<InterceptHtlcRequest, Status>>;
 
     async fn route_htlcs(
         &self,
-        request: tonic::Request<tonic::Streaming<RouteHtlcRequest>>,
+        request: tonic::Request<tonic::Streaming<CompleteHtlcResponse>>,
     ) -> Result<tonic::Response<Self::RouteHtlcsStream>, Status> {
         let mut stream = request.into_inner();
 
         // First create new channel that we will use to send responses back to gatewayd
         let (gatewayd_sender, gatewayd_receiver) =
-            mpsc::channel::<Result<RouteHtlcResponse, Status>>(100);
+            mpsc::channel::<Result<InterceptHtlcRequest, Status>>(100);
+
+        let mut sender = self.interceptor.sender.lock().await;
+        *sender = Some(gatewayd_sender.clone());
 
         // Spawn new thread that listens for events from the input stream
         let interceptors = self.interceptor.clone();
         tokio::spawn(async move {
             while let Some(res) = stream.next().await {
-                if let Ok(route_request) = res {
-                    match route_request.action {
-                        Some(route_htlc_request::Action::SubscribeRequest(subscribe_request)) => {
-                            interceptors.subscriptions.lock().await.insert(
-                                subscribe_request.short_channel_id,
-                                gatewayd_sender.clone(),
-                            );
-                        }
-                        Some(route_htlc_request::Action::CompleteRequest(complete_request)) => {
-                            let _ = Self::complete_htlc(
-                                complete_request,
-                                interceptors.clone(),
-                                gatewayd_sender.clone(),
-                            )
-                            .await
-                            .map_err(|e| {
-                                error!("CLN extension failed to complete HTLC: {:?}", e);
-                            });
-                        }
-                        None => {
-                            error!("No action was sent as part of RouteHtlcRequest");
-                        }
-                    }
+                if let Ok(complete_request) = res {
+                    let _ = Self::complete_htlc(complete_request, interceptors.clone())
+                        .await
+                        .map_err(|e| {
+                            error!("CLN extension failed to complete HTLC: {:?}", e);
+                        });
```

Comment:
I don't really like this `let _ = ...` pattern. Probably better to also do `if let Err(e) = ...` ?